src/friendly-iframe-embed.js

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {dev} from './log';
+import {extensionsFor} from './extensions';
+import {loadPromise} from './event-helper';
+import {getTopWindow, setParentWindow} from './service';
+
+
+/**
+ * Parameters used to create the new "friendly iframe" embed.
+ * - html: The complete content of an AMP embed, which is itself an AMP
+ *   document. Can include whatever is normally allowed in an AMP document,
+ *   except for AMP `<script>` declarations. Those should be passed as an
+ *   array of `extensionIds`.
+ * - extensionsIds: An optional array of AMP extension IDs used in this embed.
+ * - fonts: An optional array of fonts used in this embed.
+ *
+ * @typedef {{
+ *   html: string,
+ *   extensionIds: (?Array<string>|undefined),
+ *   fonts: (?Array<string>|undefined),
+ * }}
+ */
+export let FriendlyIframeSpec;
+
+
+/**
+ * @type {boolean|undefined}
+ * @visiblefortesting
+ */
+let friendlyIframeSupported;
+
+/**
+ * @param {boolean|undefined} val
+ * @visiblefortesting
+ */
+export function setFriendlyIframeSupportedForTesting(val) {
+  friendlyIframeSupported = val;
+}
+
+/**
+ * Returns `true` if the Friendly Iframes are supported.
+ * @return {boolean}
+ */
+export function isFriendlyIframeSupported() {
+  if (friendlyIframeSupported === undefined) {
+    friendlyIframeSupported = 'srcdoc' in HTMLIFrameElement.prototype;
```

Comment:
Could you describe it more? Do you mean we can do encoded data URI? Is this supported everywhere?